### PR TITLE
Update LastSentPosition when entity has no speed

### PR DESCRIPTION
**Pull request by peterbell10**

### CODE REVIEW

File: src/Entities/Entity.cpp [hunk 1]
```diff
@@ -1878,72 +1878,69 @@ void cEntity::TeleportToCoords(double a_PosX, double a_PosY, double a_PosZ)
...
+	else
+	{
+		// Speed is zero, send this to clients once only as well as an absolute position
+		if (!m_bHasSentNoSpeed)
 		{
-			// Movin'
 			m_World->BroadcastEntityVelocity(*this, a_Exclude);
-			m_bHasSentNoSpeed = false;
+			m_World->BroadcastTeleportEntity(*this, a_Exclude);
+			m_LastSentPosition = GetPosition();
```
Comment (peterbell10) [inline src/Entities/Entity.cpp]: This is the main change, previously `m_LastSentPosition` becomes invalid whenever an entity stops moving for even a single tick. That could cause big desyncs if the movement is very start-stop.

**Comment by peterbell10** [comment]
Oh, and the diff only makes sense if ignoring whitespace:
https://github.com/cuberite/cuberite/pull/4487/files?utf8=%E2%9C%93&diff=unified&w=1

**Comment by mathiascode** [comment]
Seems like there were quite a lot of issues that stemmed from BroadcastMovementUpdate. I'll compile and test.

**Comment by mathiascode** [comment]
Boat positions update properly in 1.9+ now, but players are still "invisible" (in the wrong location client-side) sometimes. Can be reproduced by having two players standing still next to each other, and disconnecting/reconnecting a few times.

**Comment by peterbell10** [comment]
Maybe try this + #4480?

**Comment by mathiascode** [comment]
Doesn't seem to help. You still have to fly around for a bit before the other player appears.

**Comment by madmaxoft** [comment]
Is squash & merge ok?

**Comment by peterbell10** [comment]
Yeah, that's fine I think.

**Pull request merged**